backend/endpoints/states.py

```diff
@@ -94,6 +107,14 @@ async def update_state(request: Request, id: int) -> StateSchema:
         fs_asset_handler.write_file(file=file, path=db_state.file_path)
         db_state_handler.update_state(db_state.id, {"file_size_bytes": file.size})
 
+    # Set the last played time for the current user
+    current_user = request.user
+    rom_user = db_rom_handler.get_rom_user(db_state.rom_id, current_user.id)
+    if not rom_user:
+        rom_user = db_rom_handler.add_rom_user(db_state.rom_id, current_user.id)
+    rom_user.last_played = datetime.now()
```

Comment:
I think we should save timezone-aware timezones every time we update `last_played`. Also, just to avoid some datetime discrepancies, we can create a `now` variable, and use it to both populate `last_played` and call `update_rom_user`.